composeApp/src/commonMain/kotlin/sp/bvantur/tasky/home/data/HomeRepositoryImpl.kt

```diff
@@ -40,4 +42,15 @@ class HomeRepositoryImpl(
             event.asAgendaItem()
         }
     }
+
+    override suspend fun syncPendingAgendaItems() {
+        localDataSource.getPendingAgendaItems().collect { items ->
+            println("collect")
+            val eventsToSync = items.map { event ->
+                println("collect event = $$event")
+                event.id
+            }
+            syncScheduler.scheduleAgendaSync(eventsToSync)
+        }
```

Comment:
Could it make more sense here to move the responsibility of fetching the pending items to the `syncScheduler`? This way you make it simpler to call this scheduler function, else everywhere you need to schedule a sync, you’d also need to depend on the local data source and fetch the pending items. This is known as temporal coupling.

Reply:
If I worked on a Android native project, I would agree with that, but here, the logic of how actual syncing is done will be divided between platforms. That means that it would require me to implement fetching from the database twice, or even more if we had more targets.